src/backend/Controllers/OffersController.cs

```diff
@@ -71,5 +71,22 @@ public async Task<ActionResult<CreateOfferDto>> Create([FromForm] CreateOfferDto
 
             return Ok(_mapper.Map<OfferDto>(offer));
         }
+        
+        
+        [HttpPut("{id:int}")] // "api/offers/{id}" id of the offer
+        public async Task<ActionResult<UpdateOfferDto>> Create([FromForm] UpdateOfferDto updateOfferDto, int id)
+        {
+            
+            var oldOffer = _offersService.GetById(id); 
+            
+            //var user = (User) HttpContext.Items["user"];
+            //if( oldOffer.Giver != user) 
+             //   return BadRequest("Invalid offer for this user");
+            
+            var newOffer = _mapper.Map<UpdateOfferDto>(updateOfferDto);
+            _offersService.UpdateOffer(oldOffer, newOffer);
+           
+            return Ok(_mapper.Map<OfferDto>(oldOffer));
```

Comment:
Please be more cautious. The return type of the method is `UpdateOfferDto` but you are actually returning and `OfferDto`. That's the problem of blindly rewriting one type with the other. You are actually taking in an `UpdateOfferDto` as parameter and returning a full `OfferDto`. If you look at offer create method, it takes in a `CreateOfferDto` and returns `OfferDto`. That also allows to eliminate the call to `Ok()` since it is not needed when returning a valid object.